File: src/test/java/utils/HammingTest.java
package utils;

import org.junit.Test;
import static org.junit.Assert.*;

public class HammingTest {
    @Test
    public void hammingDistanceTest() {
        byte[] fst = Codec.fromAscii("this is a test");
        byte[] snd = Codec.fromAscii("wokka wokka!!!");

        assertEquals(37, Hamming.computeDistance(fst, snd));
    }
}

File: src/test/kotlin/utils/byteops/HammingTest.kt
@file:OptIn(ExperimentalUnsignedTypes::class)

package utils.byteops

import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe
import utils.codec.Ascii
import utils.codec.toBytes

class HammingTest : FunSpec({
    test("Given test cases pass") {
        val fst = Ascii("this is a test").toBytes()
        val snd = Ascii("wokka wokka!!!").toBytes()
        fst hammingDistance snd shouldBe 37
    }
})